File: cellulosesz-api/src/main/java/top/likoslupus/cellulosesz/api/item/ItemGrantResult.java
package top.likoslupus.cellulosesz.api.item;

import static top.likoslupus.cellulosesz.api.validation.NumericChecks.requirePositive;
import static top.likoslupus.cellulosesz.api.validation.RangeChecks.requireInRange;

public record ItemGrantResult(
        int requested,
        int granted
) {

    public ItemGrantResult {
        requirePositive(requested, "requested");
        requireInRange(granted, 0, requested, "granted");
    }

    public boolean complete() {
        return requested == granted;
    }

}

File: cellulosesz-api/src/main/kotlin/top/likoslupus/cellulosesz/api/item/ItemGrantResult.kt
package top.likoslupus.cellulosesz.api.item

import top.likoslupus.cellulosesz.api.validation.requireInRange
import top.likoslupus.cellulosesz.api.validation.requirePositive

@JvmRecord
public data class ItemGrantResult(
    public val requested: Int,
    public val granted: Int
) {

    init {
        requested.requirePositive { "requested" }
        granted.requireInRange(0, requested) { "granted" }
    }

    public fun complete(): Boolean = requested == granted

}
